x-pack/plugins/security_solution/public/management/pages/policy/view/event_filters/layout/policy_event_filters_layout.tsx

```diff
@@ -15,33 +15,38 @@ import {
   EuiText,
   EuiSpacer,
   EuiLink,
+  EuiPageContent,
 } from '@elastic/eui';
-import { useAppUrl } from '../../../../../../common/lib/kibana';
+import { useAppUrl, useHttp } from '../../../../../../common/lib/kibana';
 import { APP_UI_ID } from '../../../../../../../common/constants';
 import { ImmutableObject, PolicyData } from '../../../../../../../common/endpoint/types';
 import { getEventFiltersListPath } from '../../../../../common/routing';
 import { useGetAllAssignedEventFilters, useGetAllEventFilters } from '../hooks';
 import { ManagementPageLoader } from '../../../../../components/management_page_loader';
 import { PolicyEventFiltersEmptyUnassigned, PolicyEventFiltersEmptyUnexisting } from '../empty';
+import { PolicyEventFiltersList } from '../list';
+import { EventFiltersHttpService } from '../../../../event_filters/service';
 
 interface PolicyEventFiltersLayoutProps {
   policyItem?: ImmutableObject<PolicyData> | undefined;
 }
 export const PolicyEventFiltersLayout = React.memo<PolicyEventFiltersLayoutProps>(
   ({ policyItem }) => {
     const { getAppUrl } = useAppUrl();
+    const http = useHttp();
+    const eventFiltersService = useMemo(() => new EventFiltersHttpService(http), [http]);
 
     const {
       data: allAssigned,
       isLoading: isLoadingAllAssigned,
       isRefetching: isRefetchingAllAssigned,
-    } = useGetAllAssignedEventFilters(policyItem?.id);
+    } = useGetAllAssignedEventFilters(eventFiltersService, policyItem?.id);
 
     const {
```

Comment:
nitpick: you don't need this query every time the page loads, you only need it if the `allAsigned` one came empty. you can pass the `enabled` parameter to `useQuery` to make the request trigger or not programmatically.

Reply:
Yeah I needed to get the total counter and decide if display the empty state or not